documentation/auto-advance.md

```diff
@@ -40,83 +49,96 @@ By default, if this value is not set, Dialogic uses a multiplier of 1, causing n
-- **Plain Settings**: `fixed_delay`, `per_word_delay`, `per_character_delay`, `await_playing_voice`, …
+As long as one of the enable conditions is `true`, Auto-Advance will continue.
+This allows stacking the reasons why Auto-Advance has been enabled.\
+Imagine, the player enables Auto-Advance, but the Text event forces it on until the
+next event.\
+Thanks to the multiple enable conditions, if the player disables Auto-Advances,
+it will still carry on until the next event.
 
-As long as one of the enable conditions is `true`, Auto-Advance will continue. We recommend to set them via following helper methods:
+You can turn any of the enable variables to `true` to enable Auto-Advance. If you have an Auto-Advance button, you can use the following code to enable the feature:
 
-- `Dialogic.Text.set_autoadvance_until_unread_message(is_enabled: bool)`
-- `Dialogic.Text.set_autoadvance_system(is_enabled: bool)`
-- `Dialogic.Text.set_autoadvance_until_next_event(is_enabled: bool)`
-
-If you set the enable flags without these methods, Auto-Advancing *won't* start if it's not enabled.
-These methods will automatically evaluate whether Auto-Advance shall be turned on or off.
+```gdscript
+Dialogic.Input.auto_advance.enabled_until_user_input = true
+```
 
 #### Ignored Characters
 
-These characters will be ignored when calculating the text. This feature allows you to further fine-tune your Auto-Advance experience. If you don't need it, simply toggle it off.
+These characters will be ignored when calculating the text. This feature allows
+you to further fine-tune your Auto-Advance experience. If you don't need it,
+simply toggle it off.
 
 ### Auto-Advance via BBCode
 
 BBCode tags can enable Auto-Advance via `[aa]` or `[aa = 10]`.\
 This forces Auto-Advance to last at least until the next Text Timeline Event.\
-If the BBCode provides a valid duration in seconds, Auto-Advance will wait for this amount, ignoring `per_word_delay` and `per_character_delay`, but respecting `await_playing_voice`.
+If the BBCode provides a valid duration in seconds, Auto-Advance will wait for
+this amount, ignoring `per_word_delay` and `per_character_delay`, but respecting `await_playing_voice`.
 
-Alternatively, if you want to override the Auto-Advance time for the current event only, add a `?` behind the duration: `[aa = 10?]`.
+Alternatively, if you want to override the Auto-Advance time for the current
+event only, add a `?` behind the duration: `[aa = 10?]`.
 
 ### Signal
 
-Stay informed about changes in Auto-Advance's state by leveraging the `Dialogic.Text.autoadvance_changed` signal.
+Stay informed about changes in Auto-Advance's state by leveraging the
+`Dialogic.Input.auto_advance.toggled` signal.
 
 ```gdscript
-signal autoadvance_changed(enabled: bool)
+signal toggled(enabled: bool)
 ```
 
 If you are interested in how Dialogic uses this signal internally, the `default_input_handler.gd` file connects to this signal:
```

Comment:
The default_input_handler file is removed. The input subsystem now uses the signal.